tests/Avalonia.Markup.Xaml.UnitTests/Xaml/DependsOnAttributeTest.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System.Collections.Generic;
+using Avalonia.Controls;
+using Avalonia.Metadata;
+using Avalonia.UnitTests;
+using Xunit;
+
+namespace Avalonia.Markup.Xaml.UnitTests.Xaml;
+
+public class DependsOnAttributeTest
+{
+    const string Case1_Xaml = @"<Window xmlns='https://github.com/avaloniaui'
+        xmlns:x='http://schemas.microsoft.com/winfx/2006/xaml'
+        xmlns:p='clr-namespace:Avalonia.Markup.Xaml.UnitTests.Xaml;assembly=Avalonia.Markup.Xaml.UnitTests'>
+   <p:DependsOnControlTest x:Name='target' First='1' Second='2' />
+</Window>";
+
+    const string Case2_Xaml = @"<Window xmlns='https://github.com/avaloniaui'
+        xmlns:x='http://schemas.microsoft.com/winfx/2006/xaml'
+        xmlns:p='clr-namespace:Avalonia.Markup.Xaml.UnitTests.Xaml;assembly=Avalonia.Markup.Xaml.UnitTests'>
+   <p:DependsOnControlTest x:Name='target' Second='2' First='1'  />
```

Comment:
DependsOnControlTest is not in Avalonia.Markup.Xaml.UnitTests.Xaml namespace.
It's a nested type, you need to move it out of the nested scope.